### PR TITLE
MNT: Remove redundant method, fix signature and add doc-string to `draw_tex`

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Doc-string is maybe not required as it is so close to `draw_text`, but still...

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/backends/backend_pgf.py [hunk 1]
```diff
@@ -693,9 +693,9 @@ def draw_image(self, gc, x, y, im, transform=None):
                   interp, w, h, fname_img))
         _writeln(self.fh, r"\end{pgfscope}")
 
-    def draw_tex(self, gc, x, y, s, prop, angle, ismath="TeX", mtext=None):
+    def draw_tex(self, gc, x, y, s, prop, angle, *, mtext=None):
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pgf.py:696]: Can we do this without deprecation?

Comment (oscargus) [reply lib/matplotlib/backends/backend_pgf.py:696]: Unclear. It is not used and was deprecated from all other `draw_tex` methods in #17484. However, it seems to have been forgotten for the pgf backend.

Comment (oscargus) [reply lib/matplotlib/backends/backend_pgf.py:696]: The deprecations were removed in #19900 and while it was touched, it does not seem like anyone noticed that the signature was different.

Comment (timhoffm) [reply lib/matplotlib/backends/backend_pgf.py:696]: Ok, let's try.

**Review by timhoffm — APPROVED** [review]

**Pull request merged**